File: cpp/1668_最大重复子字符串_easy.cpp
class Solution {
public:
    int maxRepeating(string sequence, string word) {
        int res = 0;
        for(int i=0; i<sequence.length(); i++){
            int index = 0;
            int ans = 0;
            if(sequence[i]==word[index]){
                int j=i;
                while(index<word.length() && j<sequence.length()){
                    if(sequence[j]==word[index]){
                        j++;
                        index++;
                    }
                    else
                        break;
                    if(index==word.length()){
                        ans++;
                        index = 0;
                    }
                }
                res = max(res, ans);
            }
        }
        return res;
    }
};

File: python/1668_最大重复子字符串_easy.py
class Solution:
    def maxRepeating(self, sequence: str, word: str) -> int:
        res = 0
        for i, ch in enumerate(sequence):
            index = 0
            ans = 0
            if ch == word[index]:
                # j, index = i+1, index+1
                j = i
                while(index<len(word) and j<len(sequence)):
                    if sequence[j] == word[index]:
                        j, index = j+1, index+1
                    else:
                        break
                    if index == len(word):
                        ans += 1
                        index = 0
                res = max(res, ans)
        return res